src/firefly/js/ui/RadiusInputFields.jsx

```diff
@@ -0,0 +1,205 @@
+         valid
+     });
+}
+
+const propTypes={
+    label:       PropTypes.string,
+    labelWidth:  PropTypes.number,
+    value:       PropTypes.string.required,
+    unit:        PropTypes.string,
+    min:         PropTypes.number.isRequired,
+    max:         PropTypes.number.isRequired
+};
+
+/**
+ * @param {string} value size value in degree
+ * @param {string} displayValue size value displayed
+ * @param {string} unit selected unit
+ * @param {bool}   valid validation of size value in degree
+ *
+ */
+
+class RadiusInputFieldView extends Component {
+    constructor(props) {
+        super(props);
+
+        this.updateState(props);
+
+        this.onSizeChange = this.onSizeChange.bind(this);
+        this.onUnitChange = this.onUnitChange.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.updateState(nextProps);
+    }
+
+    updateState(props) {
+        var unit = props.unit ? props.unit : 'deg';
+        var valid = isSizeValid(props.value, props.min, props.max);
+        var value = (valid&&props.value) ? props.value  : '';
+        var displayValue = props.displayValue ? props.displayValue : sizeFromDeg(value, unit);
+
+        this.state = {
+            value,
+            unit,
+            displayValue,
+            valid
+        };
+    }
+
+    onSizeChange(ev) {
+        var displayValue = get(ev, 'target.value');
+
+        //console.log('type '+ev.type + ' displayValue '+displayValue);
+
+        // validation test is determined when the typing is done
+        // update displayVvalue, value, and valid
+        if (ev.type.startsWith('blur') || (ev.type.startsWith('key') && ev.key === 'Enter')) {
+            var tmpDeg = sizeToDeg(displayValue, this.state.unit);
+            var valid = isSizeValid(tmpDeg, this.props.min, this.props.max);
+            var value = (valid) ? tmpDeg: '';
+            var stateUpdate = Object.assign({}, this.state, { displayValue,  value, valid });
+
+            this.setState(stateUpdate);
+            this.props.onChange(Object.assign({}, ev, {sizeInfo: stateUpdate }));
+        } else {
+            this.setState( { displayValue, valid: true });
+        }
+    }
+
+    onUnitChange(ev) {
+        var unit = get(ev, 'target.value');
+
+        // update displayValue and unit
+        if (unit !== this.state.unit) {
+            var displayValue = sizeFromDeg(this.state.value, unit);
+            var stateUpdate = Object.assign({}, this.state, {unit, displayValue});
+
+            this.setState({unit, displayValue});
+            this.props.onChange(Object.assign({}, ev, {sizeInfo: stateUpdate}));
+        }
+    }
+
+    render() {
+        var minInUnit = sizeFromDeg(this.props.min, this.state.unit);
+        var maxInUnit = sizeFromDeg(this.props.max, this.state.unit);
+        var sign = unitSign[this.state.unit];
+        var message = `Valid range between: ${minInUnit}${sign} and ${maxInUnit}${sign}`;
+        var {displayValue, valid, unit} = this.state;
```

Comment:
_Just being picky_ - you could move this line up and not have a `this.state.unit` in lines 147,148, and 149